app/org/sagebionetworks/bridge/dynamodb/DynamoTaskDao.java

```diff
@@ -72,19 +71,16 @@ public boolean taskRunHasNotOccurred(String healthCode, String runKey) {
         hashKey.setHealthCode(healthCode);
         hashKey.setRunKey(runKey);
         
-        DynamoDBQueryExpression<DynamoTask> query = new DynamoDBQueryExpression<DynamoTask>()
-            .withHashKeyValues(hashKey);
-
-        return (mapper.count(DynamoTask.class, query) == 0);
+        RangeKeyCondition rangeKeyCondition = new RangeKeyCondition("runKey").eq(runKey);
+        int count = index.queryKeyCount("healthCode", healthCode, rangeKeyCondition);
+        return (count == 0);
     }
```

Comment:
It represents a unique run of the schedule plan. All the activities in a schedule plan will share the same run key (schedule plan guid + scheduledOn time). This is how we identify if tasks from the scheduler have already been persisted in DDB, or not.